### PR TITLE
Add new test case execute guest-fstrim to guest agent

### PR DESCRIPTION
**Pull request by qingwangrh**
ID:1821543
Signed-off-by: qingwangrh <qinwang@redhat.com>

### CODE REVIEW

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=thin_provision_guest_fstrim --guestname=RHEL.8.2.0 --imageformat=qcow2 --driveformat=virtio_scsi --nicmodel=virtio_net  --platform=x86_64 --clone=no
(1/4) Host_RHEL.m8.u1.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi-block.ext4.q35: PASS (53.01 s)
 (2/4) Host_RHEL.m8.u1.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi-block.xfs.q35: PASS (51.50 s)
 (3/4) Host_RHEL.m8.u1.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi-hd.ext4.q35: PASS (51.57 s)
 (4/4) Host_RHEL.m8.u1.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi-hd.xfs.q35: PASS (50.56 s)

 (1/4) Host_RHEL.m8.u1.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi-block.ext4: PASS (51.87 s)
 (2/4) Host_RHEL.m8.u1.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi-block.xfs: PASS (50.14 s)
 (3/4) Host_RHEL.m8.u1.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi-hd.ext4: PASS (51.45 s)
 (4/4) Host_RHEL.m8.u1.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi-hd.xfs: PASS (51.35 s)

**Comment by qingwangrh** [comment]
@YongxueHong @zhenyzha Please help to review

File: qemu/tests/thin_provision_guest_fstrim.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        return output.split()[0]
+
+    def get_guest_discard_disk():
+        """
+        Get discard disk on guest.
+        """
+        if params["drive_format_%s" % data_tag] == "scsi-block":
+            return get_scsi_debug_disk(session)
+        else:
```
Comment (YongxueHong) [inline qemu/tests/thin_provision_guest_fstrim.py:53]: `else:` is redundant.

File: qemu/tests/cfg/thin_provision_guest_fstrim.cfg [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    guest_agent_serial_type = virtio
+    variants:
+        -ext4:
+            format = "ext4"
+        -xfs:
+            format = "xfs"
+    variants:
+        -scsi-block:
+            drive_format_stg1 = scsi-block
+        -scsi-hd:
```
Comment (YongxueHong) [inline qemu/tests/cfg/thin_provision_guest_fstrim.cfg:32]: Please use `scsi_hd` for `scsi-hd`  and add a space between `-` and `scsi-hd`.

File: qemu/tests/thin_provision_guest_fstrim.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    params["start_vm"] = "yes"
+    params["image_name_%s" % data_tag] = disk_name
+
+    error_context.context("Boot guest with disk '%s'" % disk_name,
+                          logging.info)
+    # boot guest with scsi_debug disk
+    env_process.preprocess_vm(test, params, env, vm_name)
+    vm = env.get_vm(vm_name)
+    vm.verify_alive()
+    timeout = float(params.get("login_timeout", 240))
```
Comment (YongxueHong) [inline qemu/tests/thin_provision_guest_fstrim.py:94]: You could use `timeout = params.get_numeric("login_timeout", 240, float)` and the` wait_for_login`'s `timeout` defaults `240`, maybe do not need to set timeout again.

Comment (qingwangrh) [reply qemu/tests/thin_provision_guest_fstrim.py]: updated

File: qemu/tests/cfg/thin_provision_guest_fstrim.cfg [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            disk_serial = TARGET_DISK1
+            blk_extra_params_stg1 += "serial=${disk_serial}"
+#    test command in guest
+    guest_mount_point = "/home/test"
+    guest_test_file = "${guest_mount_point}/test.img"
+    guest_format_command = "mkdir -p ${guest_mount_point};mkfs.${format} {0} && mount {0} ${guest_mount_point}"
+    guest_dd_command = "dd if=/dev/zero of=${guest_test_file}"
+    guest_rm_command = "rm -rf ${guest_test_file};sync"
+
+
```
Comment (YongxueHong) [inline qemu/tests/cfg/thin_provision_guest_fstrim.cfg:43]: Please only one new blank line at the end of the file.

Comment (qingwangrh) [reply qemu/tests/cfg/thin_provision_guest_fstrim.cfg]: updated

**Comment by zhenyzha** [comment]
python2 -drive
 (1/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi_block.ext4: PASS (130.88 s)
 (2/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi_block.xfs: PASS (144.77 s)
 (3/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi_hd.ext4: PASS (140.33 s)
 (4/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.ppc64le.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi_hd.xfs: PASS (116.78 s)

python3 -blockdev:
 (1/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.1.ppc64le.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi_block.ext4: PASS (129.52 s)
 (2/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.1.ppc64le.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi_block.xfs: PASS (112.35 s)
 (3/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.1.ppc64le.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi_hd.ext4: PASS (122.34 s)
 (4/4) Host_RHEL.m8.u2.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.1.ppc64le.io-github-autotest-qemu.thin_provision_guest_fstrim.scsi_hd.xfs: PASS (123.70 s)

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Pull request merged**